[18.0][ADD] account_payment_batch_oca (ex account_payment_order)

alexis-via opened the pull request.

alexis-via committed:
Start to port bank-payment to v9 (with a lot of improvements) during the Sorrento Code sprint 2016

Improvements include:
- full re-organisation of modules and big re-organisation of the code
- simplification of the code related to the fact that support for direct debit is now in t
he base module, not added by an optional module account_direct_debit (module was removed)
- new design of the wizard to select move lines to pay
- support for non-SEPA file transfer-
- support for German direct debit SEPA files (fixes bug #129)
- remove workflow of payment.order
- add wizard of selection of move lines to pay

sbidoul committed:
[FIX+IMP] account_payment_order: 2 things:

* IMP: prevent deleting a move-line that is linked to a payment order line
* FIX: better filtering of move lines already included in open payment order

alexis-via committed:
Add support for multi-currency in the generation of the account move.

Take into account all the remaining remarks of Frederic Clementi dated May 30:
- see multi-currency amounts in the wizard to add transaction lines
- add help msg on bank payment lines

pedrobaeza committed:
[IMP] account_payment_order: Remove useless migration script

yvaucher committed:
Move the account_payment_mode ir.rule in account_payment_mode module

sbidoul committed:
[FIX+IMP] account_payment_order: 2 more things:

* IMP: extensibility of _prepare_move

  Pass the list of bank lines to _prepare_move so it is possible to customize the move (eg it's name) based on the lines being paid.

* FIX: payment order: add missing onchange dependencies in move selection wizard

alexis-via committed:
[FIX] account_payment_order: 4 things:

* Fix crash when selecting a customer that has a direct debit payment mode but no mandate
* Hide many fields when payment_order_ok is False on a payment_mode
* Fix payment line communication when the move line is not linked to an invoice (problem found when porting donation_debit_debit to v9)
* For the communication field, if self.move_id.ref is False, fallback on self.move_id.name

sbidoul committed:
[IMP] set partner on offsetting account move line if possible

fixes #277

yvaucher committed:
Reactivate company rules in account_payment_order

Commit:
Port almost all modules to v10 (#305)

Port almost all modules to v10

* Update to EPC Rulebook v9.2 that start to apply on 2016-11-20 (bug #300)

Commit:
[FIX] account_payment_order: Use 'index' instead of 'select'

jalzaga committed:
Remove internal_type=other restriction for transfer account (#311)

alexis-via committed:
Creation/Modification of bank info only for Account Payment group

vrenaville committed:
[FIX] fill date instead off maturity date on account move line generated

mdietrichc2c committed:
Fix 2 issues on form view for payment order

alexis-via committed:
[FIX+IMP] account_payment_order:

* FIX: Bug #353 crash when selecting a payment mode that has a variable link to a bank account
* FIX: Add 'post_move' option on account.payment.mode with default True, to keep the previous behavior
* IMP: Explicit error when bank account is missing on bank journal
* IMP: Add field default_date_prefered on payment mode.

GillesTephaneMeyomesse committed:
[ADD] description field for payment order

mdietrichc2c committed:
[IMP] account_payment_order: Don't allow deletion of an uploaded payment order + allow to cancel an uploaded payment order

grindtildeath committed:
[IMP] account_payment_order: Restriction + tests

* Restrict the deletion of bank payment line
* Add test_cancel_payment_order

pedrobaeza committed:
[FIX] account_payment_order: Adapt test to v10

Commit:
[IMP] account_payment_order: select partner bank account automatically.

grindtildeath committed:
[10.0]Fix payment method onchange with multicompany (#374)

GillesTephaneMeyomesse committed:
[ADD] report for payment order (#370)

etobella committed:
[MIG] account_payment_order

JordiBForgeFlow committed:
When the user looks for open payables or receivables, in the
context of payment orders, she should ocus on the amount that
is due to be paid. In this method we are forcing to display both
the amount due in company and in the invoice currency.

We then hide the fields debit and credit, because they add no value.

pedrobaeza committed:
[FIX] account_payment_order: Correct domain definition on account.payment.line view

astirpe committed:
[11.0][FIX] account_payment_order: printing report

pedrobaeza committed:
[FIX] account_payment_order: Take payment mode from move (#466)

When adding to a payment order from invoices, the payment mode
considered is the one in the invoice, but once you have confirmed it,
you can't change it, so if you change the payment mode in the journal
items, it doesn't make the difference.

With this change, this is taken into account for adding to the
payment order(s) properly.

carlosdauden committed:
[FIX] account_payment_order: Add bank payment line multi-company rule

alexis-via committed:
Several small usability improvements

- when manually encoding a payment line, get company currency as default currency
- Search on payment order numbers
- Direct search on bank journal name in payment orders
- remove widget="selection" on account.payement.mode : this object is rarely modified, so widget='selection' is not a time saver, but we miss the HTML link, which can be pretty convenient to check the configuration.

etobella committed:
[FIX] report should show the invoice number, not the move name (#477)

[IMP] account_payment_order: report show the invoice number or the move name

pedrobaeza committed:
[FIX] account_payment_order: Fix FR translation

etobella committed:
Translated using Weblate (Español (España))

Currently translated at 100,0% (193 of 193 strings)

Translation: bank-payment-11.0/bank-payment-11.0-account_payment_order
Translate-URL: https://translation.odoo-community.org/projects/bank-payment-11-0/bank-payment-11-0-account_payment_order/es/

etobella committed:
[FIX] default_partner_id added on partner_bank if it is created.

Commit:
[10.0] Add filter on partners on wizard "Create Payment Lines from Journal Items" (#489)

* Add filter on partners on wizard "Create Payment Lines from Journal Items"

* Bump version number

ecino committed:
[IMP] account_payment_order: Use bank.payment.line reference in Payment Order Move Lines (#418)

Martronic-SA committed:
Fix structured communication

Adds 'structured' to the invoice_reference_type2communication_type ans the 'structured' type is added by the module itself in account_invoice but breaks the account_move_line.py as explained here: https://github.com/OCA/bank-payment/issues/495

mncoon committed:
Translated using Weblate (German)

Currently translated at 50.8% (98 of 193 strings)

Translation: bank-payment-11.0/bank-payment-11.0-account_payment_order
Translate-URL: https://translation.odoo-community.org/projects/bank-payment-11-0/bank-payment-11-0-account_payment_order/de/

sergio-teruel committed:
[12.0][mig] account_payment_order: Migrate to v12.0

pedrobaeza committed:
[MIG] account_payment_order: Finish migration to v12

* Removed deprecated YAML tests.
* Removed document dependency, as now there's no attachment dropdown.
* Switched deprecated `compute` method of res.currency.
* Restored missing field `reference_type` in invoices.
* Fixed call to `message_post`.
* Added missing description in model.

pedrobaeza committed:
[MIG] account_banking_sepa_direct_debit: Finish migration to 12.0

* Fixed tests
* Adjust code for Python dates handling
* Update copyrights

ThomasBinsfeld committed:
[REF] Account Payment Order: add Admin in security group

ThomasBinsfeld committed:
[REF] Payment Order: better performance of draft2open method
Add a norecompute in draft2open to avoid the recomputation of total_company_currency n times (where n is the number of payment lines)

Commit:
Translated using Weblate (German)

Currently translated at 44.9% (97 of 216 strings)

Translation: bank-payment-12.0/bank-payment-12.0-account_payment_order
Translate-URL: https://translation.odoo-community.org/projects/bank-payment-12-0/bank-payment-12-0-account_payment_order/de/

rven committed:
Translated using Weblate (Dutch)

Currently translated at 62.0% (134 of 216 strings)

Translation: bank-payment-12.0/bank-payment-12.0-account_payment_order
Translate-URL: https://translation.odoo-community.org/projects/bank-payment-12-0/bank-payment-12-0-account_payment_order/nl/

pedrobaeza committed:
[FIX] account_payment_order: Fix journal domain

Steps to reproduce
------------------

* Create a payment mode with "Link to Bank Account" = "Variable".
* Select several journals.
* Create a payment order with that payment mode.
* Save.
* Go to other menu and come back to payment order.
* Try to select a journal: the list is empty.

Fixes #544

ThomasBinsfeld committed:
[REF+IMP] account_payment_order: invoice addition + move generation + restrict payment modes selection on orders

- allow add to order from invoice if order exists
- payment line creation from move line
  Benefits of the batch creation of Odoo 12.0
  The method returns the created payment lines
- split generate_move method
- only payment order option on payment methods
  New option on payment methods: payment order only (unchecked by default) allowing to enforce the use of payment orders for some payment methods

Commit:
Translated using Weblate (Italian)

Currently translated at 5.5% (12 of 217 strings)

Translation: bank-payment-12.0/bank-payment-12.0-account_payment_order
Translate-URL: https://translation.odoo-community.org/projects/bank-payment-12-0/bank-payment-12-0-account_payment_order/it/

Commit:
[FIX] account_payment_order: wrong function name + report totals + report title

ThomasBinsfeld committed:
[REF] Payment Order: move all move preparation code in _prepare_move

rven committed:
[12.0] [FIX] Failing test test_account_payment_02

Commit:
Add indexes on account payment models

The fields where the indexes are added are used in searches in
account_payment_order, which becomes really slow when a database have
many lines.

pedrobaeza committed:
[FIX] account_payment_error: Avoid singleton error in allowed_journal_ids

Fixes #612

pedrobaeza committed:
[FIX+IMP] account_payment_order: Make tests resilient + inbound inheritable + SavepointCase

Previous tests did several risk operations on tests, like performing an unbound search
on payment orders, what can lead to errors if there are some data on DB (or for example
we add in the future some payment orders in demo.

This commit also split the tests for inbound payment in a base inheritable class that
can be reused in other dependant modules, and the test of this module.

Finally, we use SavepointCase for improving performance in tests.

carlosdauden committed:
[IMP] account_payment_order: Select first partner bank

Commit:
Translated using Weblate (Catalan)

Currently translated at 100.0% (223 of 223 strings)

Translation: bank-payment-12.0/bank-payment-12.0-account_payment_order
Translate-URL: https://translation.odoo-community.org/projects/bank-payment-12-0/bank-payment-12-0-account_payment_order/ca/

sergio-teruel committed:
[FIX] account_payment_order: Fix payment order report. Adapt to bootstrap4

pedrobaeza committed:
[FIX] account_payment_order: Sequences must be noupdate=1

pedrobaeza committed:
[FIX] account_payment_order: Restore colors in payment order tree view

pedrobaeza committed:
[IMP] account_payment_order: Print payment order in user lang if not generated

When you haven't still generated the file, `generated_user_id` is not filled, and
so, you can't control the language in which the payment order is going to be printed.

Adding a condition for using the user language in that case makes this more
comfortable.

Commit:
[IMP] account_payment_order: change 'bill reference' column value in reports

CarlesAntoli committed:
Added translation using Weblate (Catalan)

rven committed:
[IMP] account_payment_order: black, isort

rven committed:
[MIG] account_payment_order: Migration to 13.0

astirpe committed:
[FIX] multicompany rule

ValentinVinagre committed:
[FIX] account_payment_order: set partner_bank_id + order generation from invoice tree + attachment file name

- fix set partner_bank_id in account move lines
  Case: The bank account isn't set properly in the payment order lines, because the bank account is only got from account.move.line or the first bank account in the partner_id, but never from the invoice (account.move).
  Solution: With this fix, the bank account will be get from account.move when matching the criteria.
-  fix payment orders from invoice tree view
  Case: User can't add invoices to payment/debit orders from the tree view but it's still possible to add them one by one from the form view.
  Solution: Remove binding_views="form": by default are "tree & form". With this fix, users can add payment/debit orders from the invoice tree view
- fix the attachment file name & dowload.
  Case: After the configuration of a debit order and generating the .xml file, user can't download it from the next step screen using the download link.
  Solution: This fix makes the .xml file downloadable from the .xml file name as it was in older versions

StefanRijnhart committed:
[FIX] Set priority on alternate view

alexis-via committed:
Fix state condition on invoice view

luc-demeyer committed:
[IMP] account_payment_order: move payment order menus

ibuioli committed:
Added translation using Weblate (Spanish (Argentina))

Currently translated at 100.0% (220 of 220 strings)

Translation: bank-payment-13.0/bank-payment-13.0-account_payment_order
Translate-URL: https://translation.odoo-community.org/projects/bank-payment-13-0/bank-payment-13-0-account_payment_order/es_AR/

astirpe committed:
[FIX] avoid cache miss in compute

alan196 committed:
[MIG] account_payment_order: Migration to 14.0

CarlosRoca13 committed:
[MIG] account_payment_order: Migration to v14

joao-p-marques committed:
[FIX] account_payment_order: Fix tests

On multi-company settings, we need to make sure we are searching and creating objects with the correct company associated.

TT28423

Yvesldff committed:
Added translation using Weblate (French (France))

Currently translated at 86.8% (192 of 221 strings)

Translation: bank-payment-14.0/bank-payment-14.0-account_payment_order
Translate-URL: https://translation.odoo-community.org/projects/bank-payment-14-0/bank-payment-14-0-account_payment_order/fr_FR/

joao-p-marques committed:
[FIX] account_payment_order: Fix tests

After https://github.com/odoo/odoo/commit/fcaa54939e9a4f0dd5e47cd0ccffe7aa24bd451c, we need to specify the invoice date for vendor bills to be able to post it

Yvesldff committed:
Translated using Weblate (French (France))

Currently translated at 91.4% (202 of 221 strings)

Translation: bank-payment-14.0/bank-payment-14.0-account_payment_order
Translate-URL: https://translation.odoo-community.org/projects/bank-payment-14-0/bank-payment-14-0-account_payment_order/fr_FR/

pedrobaeza committed:
[FIX] account_payment_order: Computed non-stored fields needs always a value

We should assign a value for all records of the set.

Commit:
Added translation using Weblate (Spanish (Chile))

Currently translated at 100.0% (221 of 221 strings)

Translation: bank-payment-14.0/bank-payment-14.0-account_payment_order
Translate-URL: https://translation.odoo-community.org/projects/bank-payment-14-0/bank-payment-14-0-account_payment_order/es_CL/

Commit:
Translated using Weblate (Spanish)

Currently translated at 100.0% (221 of 221 strings)

Translation: bank-payment-14.0/bank-payment-14.0-account_payment_order
Translate-URL: https://translation.odoo-community.org/projects/bank-payment-14-0/bank-payment-14-0-account_payment_order/es/

baimont committed:
[13.0][IMP] account_payment_order - move date should be bank line date when offsetting_account is bank_account

bosd committed:
Translated using Weblate (Dutch)

Currently translated at 70.4% (155 of 220 strings)

Translation: bank-payment-14.0/bank-payment-14.0-account_payment_order
Translate-URL: https://translation.odoo-community.org/projects/bank-payment-14-0/bank-payment-14-0-account_payment_order/nl/

pedrobaeza committed:
[FIX] account_payment_order: Remove old migration scripts

joao-p-marques committed:
[FIX] account_payment_order: Change security group category + Make post move field invisible in payment modes + fix account move views

- Change security group category
  Before this change, the group was creating a new section in each user's settings called "Other" with a field called "Other Extra Rights" that could select "Accounting / Payments" as an option
  This places it in the correct location
  TT29182
- Make post move field invisible in payment modes
  When the generate_moves option is not selected, the post_move option should not be available
  TT29182
- fix account move views
  Current spec was causing weird display in account move view, both for customer invoices and vendor bills
  Changing the placement and adding back the label fixes the issue
  TT29027

manuelregidor committed:
[IMP]account_payment_order: payment date when adding transaction line

alexis-via committed:
[IMP] account_payment_order: computation of sepa + remove done state + UI adjustments

- Improve computation of sepa on account.payment.order: check IBAN is in SEPA zone
  Update move line generation to get transfer account from bank journal
  Update payment mode configuration accordingly (3 fields removed)
  Several improvements in payment order tree and form view
- Remove done state (migration script provided)
- Don't set reference_type field on account.move required=True, because it
  causes a lot of problems in other modules that don't depend on this one.
  Setting it as required in the view is enough.
- add optional="hide" on payment mode in invoice tree view
- FIX crash when communication=null on payment line
- payment_mode_id readonly when state != draft

AmmarOfficewalaSerpentCS committed:
[IMP] account_payment_order: Add hook method for finding previous payment orders + Pass single move in a get_account_payment_domain mehtod to find move in a method.

iammiracle committed:
[FIX] account_payment_order: Bind report to model

joao-p-marques committed:
[FIX] account_payment_order: Fix multi-company issues in tests

Use base `account` tests from Odoo to avoid issues with the environment.

ibuioli committed:
Translated using Weblate (Spanish (Argentina))

Currently translated at 100.0% (212 of 212 strings)

Translation: bank-payment-14.0/bank-payment-14.0-account_payment_order
Translate-URL: https://translation.odoo-community.org/projects/bank-payment-14-0/bank-payment-14-0-account_payment_order/es_AR/

simonstromb committed:
Added translation using Weblate (Swedish)

Currently translated at 65.0% (138 of 212 strings)

Translation: bank-payment-14.0/bank-payment-14.0-account_payment_order
Translate-URL: https://translation.odoo-community.org/projects/bank-payment-14-0/bank-payment-14-0-account_payment_order/sv/

misern2 committed:
[MIG] account_payment_order: Migration to 15.0

celm1990 committed:
[IMP] account_payment_order: outbound inheritable + SavepointCase
This commit split the tests for outbound payment in a base inheritable class that
can be reused in other dependant modules, and the test of this module.

similar to https://github.com/OCA/bank-payment/commit/86bd1a2525d45f4a2eb9146b4a0ef679068d3649

pedrobaeza committed:
[FIX] account_payment_order: Proper indentation for test + make it work

The test introduced in 7bdb286aa1a3d5beb8ee95193ed6eb4abc67ce1e is not
valid, as its indentation made it to not be executed. Anyways, the
content and the tested things are also not correct.

alexis-via committed:
account_payment_order: communication required in form view

MiquelRForgeFlow committed:
[IMP] account_payment_order: don't requiere reference_type anymore

driehuis committed:
[FIX] account_payment_order: Fallback to payment_ref if ref is undefined

juliap98 committed:
Translated using Weblate (Catalan)

Currently translated at 100.0% (212 of 212 strings)

Translation: bank-payment-15.0/bank-payment-15.0-account_payment_order
Translate-URL: https://translation.odoo-community.org/projects/bank-payment-15-0/bank-payment-15-0-account_payment_order/ca/

rousseldenis committed:
[IMP] account_payment_order: Better communication field management

- Add a hook method to retrieve communication type and communication
- Improve normal communication if there is a credit note
  If there is a credit note that partially cancel an invoice, the payment communication
  should be the combination of the invoice reference and the credit note one.
- Remove not needed assert as parameter is required
- Use the 'payment_reference' field if filled in
- Add existing payment references to communication
  If some movements have been reconciled with the original invoice,
  their references should be added in communication too.
  e.g.: Manual credit notes
- Don't duplicate communication reference

etobella committed:
[IMP] account_payment_order: Promote to Mature

pedrobaeza committed:
[FIX] account_payment_order: Don't fail due to tz

When the test hour is around midnight, and demo data using CET/CEST
timezones, current code fails due to some asserts not having this into
consideration. This fix avoids such problem, and irons the execution
no matter the hour.

ramiadavid committed:
[FIX] account_payment_order: Fix tests

Some tests need to be run on post-install
The bank_journal.inbound_payment_order_only field is true only if all inbound_payment_method_ids are payment_order_only True. For this we must include the "SEPA Direct Debit for customers" in test

alexis-via committed:
[FIX] account_payment_order: fix action returned by create_account_payment_line()

create_account_payment_line() is supposed to returned a form view for a
single payment order and a tree view for multiple payment orders. Before
this fix, it would return a tree view if you were payment more than 1
invoices while generating a single payment order, because the list
result_payorder_ids would contain several time the same ID.

Commit:
[FIX] Restore payment order number in account move line label.

wpichler committed:
[MIG] account_payment_order: Migration to 16.0

manuelregidor committed:
[IMP]account_payment_order: Add activities to account.payment.order model

pedrobaeza committed:
[REF+IMP] account_payment_order: Use native payments

The previous approach creates manually the journal entries and does all
the hard work, plus not being 100% compatible with the bank statement
reconciliation widget (requiring a patch on OCB to see blue lines).

That decision made sense on the moment it was done (v9), where the
native payment model (account.payment) was very limited, and wasn't able
to store all the needed information for the bank transaction.

Now that the limitations are gone, we can get rid off this extra model,
and generate instead `account.payment` records, using both the native
model + methods to perform the same operations.

This serves also to workaround the problem found in #966.

All the code, views and tests of main module have been adapted to this
new approach in this commit. Later commits will adapt the rest of the
modules of the suite, and add migration scripts to transit from the
previous approach to this new one.

TT39832

alexis-via committed:
[IMP] account_payment_order: add link to payment order in chatter

antony-psc committed:
[FIX] account_payment_order: Error in Batch Payments when deleting a Batch Content line

celm1990 committed:
[FIX] account_payment_order: change result_payorder_ids to list to show action correctly

Commit:
[IMP] account_payment_order: Concatenate all UserError messages
Gather all the errors to only show one with all the problem on the current debit order. This will avoid for the users to have to make multiple time the same process.

wpichler committed:
[I18N] account_payment_order: Better translations for de_DE

dannyadair committed:
Add pre_init_hook to add computed column

Add partner_bank_id column to avoid MemoryError on an existing Odoo
instance with lots of data.

partner_bank_id on account.move.line requires payment_order_ok to be True
which it won't be as it's newly introduced - again nothing to compute.
(see AccountMoveLine._compute_partner_bank_id() in models/account_move_line.py)

pedrobaeza committed:
[FIX] account_payment_order: Do correctly the fix on tests

Previous patch by Alexis created a co-dependency between
account_payment_order and account_banking_sepa_direct_debit, which is
not correct.

This patch avoids such problem and fix properly the test, although
the utility of some of them is debatible.

Commit:
[UPD] Update account_payment_order.pot

OCA-git-bot committed:
[UPD] README.rst

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: bank-payment-16.0/bank-payment-16.0-account_payment_order
Translate-URL: https://translation.odoo-community.org/projects/bank-payment-16-0/bank-payment-16-0-account_payment_order/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 5.4% (11 of 202 strings)

Translation: bank-payment-16.0/bank-payment-16.0-account_payment_order
Translate-URL: https://translation.odoo-community.org/projects/bank-payment-16-0/bank-payment-16-0-account_payment_order/it/

Reyes4711-S73 committed:
[FIX] account_payment_order: create lines with same sequence name

sysadminmatmoz committed:
Translated using Weblate (Slovenian)

Currently translated at 17.8% (36 of 202 strings)

Translation: bank-payment-16.0/bank-payment-16.0-account_payment_order
Translate-URL: https://translation.odoo-community.org/projects/bank-payment-16-0/bank-payment-16-0-account_payment_order/sl/

OCA-git-bot committed:
account_payment_order 16.0.1.0.1

Commit:
[UPD] Update account_payment_order.pot

OCA-git-bot committed:
account_payment_order 16.0.1.0.2

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: bank-payment-16.0/bank-payment-16.0-account_payment_order
Translate-URL: https://translation.odoo-community.org/projects/bank-payment-16-0/bank-payment-16-0-account_payment_order/

Nikul-OSI committed:
[FIX] Fixed Payment Transaction view

pedrobaeza committed:
[FIX] account_payment_order: treet2 > street2

Commit:
[UPD] Update account_payment_order.pot

OCA-git-bot committed:
account_payment_order 16.0.1.1.0

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: bank-payment-16.0/bank-payment-16.0-account_payment_order
Translate-URL: https://translation.odoo-community.org/projects/bank-payment-16-0/bank-payment-16-0-account_payment_order/

pedrobaeza committed:
[FIX] account_payment_order: Set payment method line for account.payment

On v15+, a new model account.payment.method.line is introduced for
fine-graining the outstanding account, being `payment_method_id` a
related field referred to this new model, so we need to change the
previous approach to select the proper method line only if found. If not
found, it will be auto-selected by standard code.

TT43278

OCA-git-bot committed:
account_payment_order 16.0.1.1.1

Gelo-fl committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (204 of 204 strings)

Translation: bank-payment-16.0/bank-payment-16.0-account_payment_order
Translate-URL: https://translation.odoo-community.org/projects/bank-payment-16-0/bank-payment-16-0-account_payment_order/es/

sanrav committed:
Translated using Weblate (French)

Currently translated at 95.0% (194 of 204 strings)

Translation: bank-payment-16.0/bank-payment-16.0-account_payment_order
Translate-URL: https://translation.odoo-community.org/projects/bank-payment-16-0/bank-payment-16-0-account_payment_order/fr/

luc-demeyer committed:
[IMP] account_payment_order: add payment lines smart button on invoice

remi-filament committed:
Translated using Weblate (French)

Currently translated at 100.0% (204 of 204 strings)

Translation: bank-payment-16.0/bank-payment-16.0-account_payment_order
Translate-URL: https://translation.odoo-community.org/projects/bank-payment-16-0/bank-payment-16-0-account_payment_order/fr/

Yadier-Tecnativa committed:
[FIX] account_payment_order Allow to download file in edit mode

OCA-git-bot committed:
account_payment_order 16.0.1.2.0

victoralmau committed:
[IMP] account_payment_order: Add a custom message if the invoice does not have a payment method

The message will be displayed when trying to create a line of a payment order to better clarify the reason.

TT44551

OCA-git-bot committed:
account_payment_order 16.0.1.2.1

pedrobaeza committed:
[FIX+IMP] account_payment_order: Better error messages on "Add to order"

- The filter for payment order lines should be applied after checking
  the lack of payment mode.
- Added message when the payment mode is not valid for payment orders.
- Added message specifying the payment order where the line is already
  present.

TT44762

Commit:
[UPD] Update account_payment_order.pot

OCA-git-bot committed:
account_payment_order 16.0.1.2.2

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: bank-payment-16.0/bank-payment-16.0-account_payment_order
Translate-URL: https://translation.odoo-community.org/projects/bank-payment-16-0/bank-payment-16-0-account_payment_order/

OCA-git-bot committed:
[UPD] README.rst

Commit:
[UPD] Update account_payment_order.pot

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: bank-payment-16.0/bank-payment-16.0-account_payment_order
Translate-URL: https://translation.odoo-community.org/projects/bank-payment-16-0/bank-payment-16-0-account_payment_order/

OCA-git-bot committed:
account_payment_order 16.0.1.2.3

OCA-git-bot committed:
[UPD] README.rst

Ivorra78 committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (206 of 206 strings)

Translation: bank-payment-16.0/bank-payment-16.0-account_payment_order
Translate-URL: https://translation.odoo-community.org/projects/bank-payment-16-0/bank-payment-16-0-account_payment_order/es/

victoralmau committed:
[FIX] account_payment_order: Define the value of the communication field correctly

- Purchases: Ref field from invoice always.
- Sales: Payment_ref or ref field from invoice.

TT45153

victoralmau committed:
[IMP] account_payment_order: Add payment_ref field to account.payment tree and form views

Co-authored-by: luc-demeyer

TT45153

Ivorra78 committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (206 of 206 strings)

Translation: bank-payment-16.0/bank-payment-16.0-account_payment_order
Translate-URL: https://translation.odoo-community.org/projects/bank-payment-16-0/bank-payment-16-0-account_payment_order/es/

victoralmau committed:
[FIX] account_payment_order: Add extra form view only to avoid open move lines in edit mode

TT45211

victoralmau committed:
[FIX] account_payment_order: Add colspan=2 to a correct visualization

OCA-git-bot committed:
[BOT] post-merge updates

alexis-via committed:
Add warning when payment order is not SEPA

The field 'sepa' on account.payment.order is only display for SEPA
payment methods.
If the option "show warning if not SEPA" is enabled on the payment
method, a warning banner is now displayed on payment orders with a SEPA
payment method which are not SEPA.

OCA-git-bot committed:
[BOT] post-merge updates

Commit:
[UPD] Update account_payment_order.pot

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: bank-payment-16.0/bank-payment-16.0-account_payment_order
Translate-URL: https://translation.odoo-community.org/projects/bank-payment-16-0/bank-payment-16-0-account_payment_order/

Commit:
[UPD] Update account_payment_order.pot

OCA-git-bot committed:
[BOT] post-merge updates

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: bank-payment-16.0/bank-payment-16.0-account_payment_order
Translate-URL: https://translation.odoo-community.org/projects/bank-payment-16-0/bank-payment-16-0-account_payment_order/

luc-demeyer committed:
[IMP]grouped invoices: " - ".join in stead of "-".join

OCA-git-bot committed:
[BOT] post-merge updates

jakobkrabbe committed:
Translated using Weblate (Swedish)

Currently translated at 60.8% (126 of 207 strings)

Translation: bank-payment-16.0/bank-payment-16.0-account_payment_order
Translate-URL: https://translation.odoo-community.org/projects/bank-payment-16-0/bank-payment-16-0-account_payment_order/sv/

Ivorra78 committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (207 of 207 strings)

Translation: bank-payment-16.0/bank-payment-16.0-account_payment_order
Translate-URL: https://translation.odoo-community.org/projects/bank-payment-16-0/bank-payment-16-0-account_payment_order/es/

alexis-via committed:
Large refactoring/improvement/cleanup of OCA/bank-payment

NEW FEATURES
- activate 'in_payment' payment_state on invoices... at last !
- payment order numbering: add ability to have a sequence specific to a particular payment mode (optional)
- add show account number policy "first_last" to show N first and N last characters of bank account number

SMALL IMPROVEMENTS
- account.payment.method: rename field "payment_order_only" to "payment_order_ok" for naming consistency with the field of account.payment.mode. Auto-set "payment_order_ok" of payment mode from the equivalent field of payment method. Migration scripts provided for account.payment.method and the data of SEPA credit transfer and SEPA direct debit.
- payment order attachment: add M2O payment_file_id that points to the payment file. No more need for a simplified attachment form view.
- keep the spaces in the scrambled bank account number, so that the user easily understands that it is a "scrambled" bank account number
- add field acc_number_scrambled on res.partner.bank for easy and direct use of scrambled account number

SIMPLIFICATION / CODE CLEANUP
- remove field partner_bank_id on account.move.line
- move code for scrambled account number from qweb report to res.partner.bank class
- remove boolean field show_bank_account_from_journal from account.payment.mode: we now consider that this option is always true i.e. when account_payment_partner is installed, Odoo takes the bank accounts to display on the customer invoice from the payment mode (if the payment mode is empty, it displays all the bank accounts of the company). As a consequence, hide partner_bank_id on customer invoice form view.
- remove fields inbound_payment_order_only and outbound_payment_order_only from account.journal : these 2 fields were not used!
- add check_company=True where it was missing and remove code of constraint that checks company consistency
- re-organise form view of payment order to equilibrate between left column and right column

TECHNICAL:
- replace @api.onchange by computed readonly=False store=True fields
- remove (or reduce) the use of demo data in tests
- speed-up tests by using tracking_disable=True
- Replace (6, 0, []) and (0, 0, {}) by Command.set() and Command.create()
- add sql unicity constraint on payment order number per company

Commit:
[UPD] Update account_payment_order.pot

OCA-git-bot committed:
[BOT] post-merge updates

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: bank-payment-16.0/bank-payment-16.0-account_payment_order
Translate-URL: https://translation.odoo-community.org/projects/bank-payment-16-0/bank-payment-16-0-account_payment_order/

Ivorra78 committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (209 of 209 strings)

Translation: bank-payment-16.0/bank-payment-16.0-account_payment_order
Translate-URL: https://translation.odoo-community.org/projects/bank-payment-16-0/bank-payment-16-0-account_payment_order/es/

alexis-via committed:
Add support for regulatory reporting in PAIN files

Harmonize values for reference_type on account.move and communication_type on account.payment.line:
2 possible values : free and structured (migration script provided)
Simplify code for reference.

mourad-ehm committed:
[16.0] FIX account_payment_order communication full

alexis-via committed:
[FIX] always exclude cancel moves in account.payment.line.create wizard

alexis-via committed:
account_payment_order: remove view inheritance that is not needed any more

alexis-via committed:
[FIX] account_payment_order: compute of bank account on payment line

Only copy bank account from move line to payment line for payment
orders, not for debit orders.

alexis-via committed:
[IMP] account_payment_order: get d/l pop-up when generating file

When the user clicks on the button "Generate payment file" in the form
view of payment order, it gets a pop-up to save the file.

alexis-via committed:
[IMP] add related field payment_method_code on account.move and account.payment.order

This new field payment_method_code on account.move and
account.payment.order is available as invisible in tree+form view: it
allows to easily show/hide fields specific to a payment method in format-specific modules.
Remove field sepa_payment_method (replaced by generic field payment_method_code)
Update field definition of payment_mode_id on account.move: precompute=True and move domain from view to field definition

alexis-via committed:
Improve link between mandate and bank accounts

If we use a payment order with mandate required, mandate is placed
before bank account (more logic) and bank account becomes readonly: it
is set by the mandate.

alexis-via committed:
[IMP] account_payment_order: set no_debit_before_maturity to True by default

alexis-via committed:
[IMP] account_payment_order: set default_date_prefered to "due" by default on payment modes for inbound payment methods with debit orders

alexis-via committed:
[FIX] account_payment_order: access rights issues on invoice

alexis-via committed:
account_payment_order: add fields needed to make account_reconcile_payment_order work

alexis-via committed:
[IMP] account_payment_order: pre-commit auto fixes

alexis-via committed:
[MIG] account_payment_order to v18

alexis-via committed:
[UDP] account_payment_order: new selectable field on account.payment.method.line

alexis-via committed:
[UDP] account_payment_order: adapt to changes in account_payment_base_oca

alexis-via committed:
[FIX] account_payment_order: update tests to reflect changes in account_payment_base_oca

alexis-via committed:
[IMP] account_payment_order: search on partner_id from account.payment.order

alexis-via committed:
[FIX] account_payment_order: Mark invoices as in_payment even when account.payment is not linked to a move

Fix name of account.payment
Adapt to changes in account_payment_base_oca

alexis-via committed:
Rename account_payment_order to account_payment_batch_oca

Update manifest and README
Add missing summary key in manifest

alexis-via committed:
[IMP] account_payment_order: many improvements/cleanups listed below

- introduce object account.payment.lot, which is useful both for SEPA
  payments but also for bank statement reconciliation

- Remove qweb report, because we don't use it in real life

- dedicated default sequence for debit orders

- remove the wizard account.invoice.payment.line.multi (create payment order from invoice) which doesn't have any field. Instead, we execute the code directly.

- replace the smart button "Journal Entries" in the button box of the form view of payment orders (which is broken anyway since the move to account.payment) by 2 smart buttons: one for account.payment (useful to have the payments in full screen) and one for invoices.

- remove field generated_user_id on account.payment.order. That field was useful in the very old days when there was no chatter, but today it is not useful any more.

- remove field payment_line_date on account.payment because it is completely useless ; the "date" field of account.payment has the date of the grouped payment lines.

- improve strings: in particular, replace "Move Lines" by "Journal Items" and "Payments Orders" by "Payment/Debit Orders".

- use _read_group() to improve perf

- field 'amount_company_currency' on account.payment.line is now stored

- remove useless method action_uploaded_cancel() on account.payment.order.

- display total_company_currency on payment order form view

- display account.payment on canceled payment orders.

- update and improve tests

-

alexis-via committed:
[IMP] account_payment_batch_oca: Block outbound payment orders when some bank accounts are untrusted.

Add warning banner when bank accounts are untrusted and a smart button
to list them.
With this mecanism in place, we can restore the native behavior for
ir.model.access for bank accounts and banks and let partner manager group
create/update them. So we don't inherit the native ACLs any more.
Remove unnecessary string "There's at least one validation error:" in
error message upon confirmation of payment order.
Avoid duplicate error message (can happen if the same untrusted bank account is
present in several payment lines).

alexis-via committed:
[IMP] Add read ACL for auditor

sbidoul commented on account_payment_batch_oca/readme/INSTALL.md:
```diff
@@ -0,0 +1,7 @@
+- base_iban
+- document
+
+This modules is part of the OCA/bank-payment suite.
```
Remove this INSTALL.md file.

alexis-via replied:
done

sbidoul commented on account_payment_batch_oca/readme/USAGE.md:
```diff
@@ -0,0 +1,11 @@
+Customers \> Debit Orders and then select the move lines to debit.
+
+This module also adds an action *Add to Payment Order* on supplier
+invoices and *Add to Debit Order* on customer invoices.
```
Does this Add to Payment Order button on vendor bill still exist? I don't see it.

sbidoul replied:
My bad, the button is visible as soon as a Payment Method is set on the invoice.

sbidoul commented on account_payment_batch_oca/models/account_payment_line.py:
```diff
@@ -0,0 +1,342 @@
+        ]
+        return res
+
+    def payment_line_hashcode(self):
```
```suggestion
    def _payment_line_hashcode(self):
```

sbidoul commented on account_payment_batch_oca/models/account_payment_line.py:
```diff
@@ -0,0 +1,342 @@
+        for field in self._lot_grouping_fields():
+            values.append(str(self[field]))
+        hashcode = "-".join(values)
+        return hashcode
```
Is it not better to return a tuple for these hash codes. This would be more robust in case there are `-` in some field values.

sbidoul commented on account_payment_batch_oca/models/account_payment_line.py:
```diff
@@ -0,0 +1,342 @@
+        values.append(str(self.move_line_id.account_id or False))
+        # Don't group the payment lines that use a structured communication
+        # otherwise it would break the structured communication system !
+        if self.communication_type != "free":
+            values.append(str(self.id))
```
```suggestion
        if self.communication_type != "free":
            values.append(str(self.id))
        else:
            values.append("")
```

So we have the same number of `-` in all grouping keys.

alexis-via replied:
Now that we use tuples, there is no more "-" in the grouping keys

sbidoul replied:
Yes but the tuples should always have the same size, so let's add None?

alexis-via committed:
[IMP] account_payment_batch_oca: add Invoice button on payment lines (Fixes issue #16)

Move payment file prepare method from account_payment_sepa_base to
account_payment_batch_oca.

Raise error if bank account doesn't have the "Send money" option only
for outgoing payment orders

alexis-via committed:
[IMP] account_payment_batch_oca: change string "payment mode" to "payment method"

Fixes #18

alexis-via committed:
[FIX] account_payment_batch_oca: fix for tests

alexis-via committed:
[UDP] account_payment_batch_oca: remove INSTALL.md

alexis-via committed:
[IMP] account_payment_batch_oca: payment_line_hashcode() switched to private

alexis-via committed:
[IMP] account_payment_batch_oca: Replace hashcodes by tuples

alexis-via committed:
[IMP] account_payment_batch_oca: payment group tuple now always has the same size

etobella reviewed (commented):
@alexis-via you should import the original commits in a different way. In the next migration, all the logic will be lost. Use filter-branch.

```
git filter-branch -f --index-filter '
git ls-files --stage | \
sed "s:ORIGINAL_NAME/:NEW_NAME/:" | \
GIT_INDEX_FILE=$GIT_INDEX_FILE.new \
git update-index --index-info && \
mv "$GIT_INDEX_FILE.new" "$GIT_INDEX_FILE"
' HEAD
```

sbidoul commented:
@etobella personally I'm not a fan of rewriting commit histories.

I think we can simply configure this repo to create new branches by marking modules installable=False like we do on mis-builder and some other repos.

etobella commented:
IMO it is more confusing to leave modules non installable. It is a mess, because people goes there, they find the module and when they try it it is not installable and they complain. The only exception is mis builder because a it has a single and involved maintainer.

In all other places, we ask for it. Should this repository be different than the others?

The reason to do it is that it will keep history in next version. Otherwise, it will seem that alexis did all the job of the module, but he took a module were a lot of people has worked on. I think we should respect that.

alexis-via commented:
@etobella I'm not a git expert and I don't understand the problem you're pointing. But I see that the commit history is present in this PR since v9 i.e. since the module name account_payment_order was introduced. It seems to me that the commit history is properly kept here, with the original module names.
I don't have a strong opinion on installable=False vs format-patch and I don't want to open the debate now ; for me, this question will arise when we'll do the migration to v19.
I am aware that, if we adopt format-patch, we will have to make sure that, when we do the format-patch for the v18 to v19 migration, we don't forget the filter on the current module name AND the old module name:

```
git format-patch --keep-subject --stdout origin/19.0..origin/18.0 -- account_payment_order account_payment_batch_oca | git am -3 --keep
```

With this command, it should keep all the commit history since v9 in the migration to v19.

pedrobaeza commented:
The problem will be with version 20 bringing the commit history. You will put the same command, and only the commits with the new name will be fetched, having the first one at your name as the one adding all the files.

For avoiding this, the renaming should be done an the whole commit history using this command:

```
git filter-branch --tree-filter 'if [ -d account_payment_order ]; then mv account_payment_order account_payment_batch_oca; fi' HEAD
git rebase origin/18.0
git push -f
```

sbidoul commented:
Ok, I think we can still worry about that when doing the next migration.

/ocabot merge nobump

etobella commented:
@sbidoul next migration will be too late. I am really surprised by this attitude of ignoring other peoples comments.

With your action, a lot of contributions will be lost (mine and from other people).  By doing this, it is clear that alexis is the only contributor of it. At the end, we are reseting history and leaving him as sole contributor...

I think we could do the same on other module where you have collaborated. For example, I think that I can take mis-builder and create a fork called report-builder inside OCA and leave it there as we did with bank-payment-alternative. Also, when doing this, I will remove your commits history. I hope it is fine for you, as you agreed in this aberration and opened the door for this kind of forks inside OCA.

I thought we were a community, but right now I don't believe on it anymore. This way of doing disgusts me and make me think if I should publish my code in OCA. I am not a perfect contributor for sure and I cannot teach lessons about it, but I am sure that this is not the way.

pedrobaeza commented:
Stéphane, now there's no solution unless you forced push in the main branch. I don't get why now such hurry while I have put the exact steps for fixing the problem, dedicating time to write it correctly and exactly.

etobella commented:
Do not worry @pedrobaeza they merged 2 PRs with 50 commits each to make it easy for us.

IT is clear what they want to achieve. Silencing others.

sbidoul commented:
@etobella @pedrobaeza so far the commit history is fully preserved, no? The problem you are mentioning needs attention when doing the next migration, so we just need to be careful then and chose one of the 4 different techniques proposed here.

etobella commented:
No, on next migration rewriting will be harder. But do not worry. It is clear that others people opinion and experience is not relevant anymore in OCA.

sbidoul commented:
Oh please, do not dramatize, Enric. I do respect your opinion. All I'm saying is that in this specific case we can postpone the debate on how to handle the history.

pedrobaeza commented:
Stéphane, I think you performed without consensus an action that can't be fixed easily, even with the warning of two people, and more when we have explained the solution, so it's serious. And there were no hurry for this. I'm also surprised that you specifically not being decided with which alternative to take, are very aligned here and with rush.

etobella commented:
I am trying to explain my point of view. You should try to understand it. Maybe I am exaggerated, but if I show an extreme point of view it is more clear.

In any case, it is clear that I am tired of this topic and it makes me think if i want to keep collaborating with the path that OCA is taking. It is not not a threat, just a feeling. Lately, the results do not compensate for the effort IMO.

rousseldenis commented:
@etobella I personally think we should refrain some extremes in debates - even more from great contributors (the thing with 'more power, more responsibilities' :-) ).

IMHO, and it is not a partisan one, @sbidoul approach is quite down-to-earth and allows to move forward and tackle the history in a near future.

I'm also very surprised by 'the path OCA is taking'. Apart this topic, I need to know (not here) objectively, and you should come with figures, what you mean. Because, if there are under the radar problems, they should be escalated and addressed.

Thanks

etobella commented:
@rousseldenis I understand that we shouldn't get in extremes, but this topic has rebased some extremes long time ago... The community is creating the first fork inside OCA. A door that should be avoided by a lot of reasons was opened. 

- We warned about it and we were ignored. 
- We had a meeting about the topic between PSCs and our hand for help was ignored. 
- As PSCs we agreed on a set of names to avoid confusions. It was ignored.
-  We made a meeting with some board members and they agreed on talking about this topic and thinking about stoping it as the idea of first fork inside OCA was not the best outcome. Next day, it was merged in front of our faces expecting us to clap on the amazing achievement.
- We warned about some ways of doing that should be improving non blocking the work of the team and we were ignored....

I don't know, I might be extreme, but I am tired of fighting on this. I want to raise my voice because it shouldn't happen this way. If you want, we kindly ask for people to step up on us again. That is what we should expect?

I have been around OCA for 8 years and it is the first time where I have been treated this way repeatedly.

pedrobaeza commented:
I totally agree with Enric, and felt the same way, with our work, offering and voice being totally despised.
